### PR TITLE
SPLICE-2249 Allow schema to be passed in JDBC connection string

### PR DESCRIPTION
**Pull request by wb14123**

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java [hunk 1]
```diff
@@ -477,7 +481,9 @@ protected SchemaDescriptor initDefaultSchemaDescriptor() throws StandardExceptio
         */
         if(cachedInitialDefaultSchemaDescr==null){
             DataDictionary dd=getDataDictionary();
-            String authorizationId=getSessionUserId();
+            if (defaultSchema != null) {
+                return dd.getSchemaDescriptor(defaultSchema, getTransactionCompile(), true);
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java]: I think we need to set cachedInitialDefaultSchemaDescr to the returned schema descriptor before returning from the function.

File: splice_machine/src/test/java/com/splicemachine/jdbc/JdbcApiIT.java [hunk 2]
```diff
@@ -127,4 +112,31 @@ public void testTimeoutControl() throws Exception {
         }
     }
 
+    @Test
+    public void testUrlWithSchema() throws Exception {
+        String url = "jdbc:splice://localhost:1527/splicedb;create=true;user=%s;password=%s;schema="
+                + CLASS_NAME;
```
Comment (yxia92) [inline splice_machine/src/test/java/com/splicemachine/jdbc/JdbcApiIT.java]: I think the attribute "create=true" should not be put in the test case, it is used the very first time to create the database.

File: splice_machine/src/test/java/com/splicemachine/jdbc/JdbcApiIT.java [hunk 3]
```diff
@@ -127,4 +112,31 @@ public void testTimeoutControl() throws Exception {
...
+            try (Statement statement = connection.createStatement()) {
+                statement.executeQuery("select * from a");
+            }
+        }
+    }
+
+    @Test
+    public void testUrlWithNonExistSchema() throws Exception {
+        String url = "jdbc:splice://localhost:1527/splicedb;create=true;user=%s;password=%s;schema="
+                + "nonexist";
```
Comment (yxia92) [inline splice_machine/src/test/java/com/splicemachine/jdbc/JdbcApiIT.java]: remove "create=true" here too.

**Comment by wb14123** [comment]
@yxia92 Fixed the issues.

**Pull request merged**